src/influx_udp_worker.erl

```diff
@@ -146,61 +215,189 @@ to_key(N) when is_integer(N) ->
 
 encode_1_test() ->
-  Data = decode_data(encode(test, [ [{1, 1}], [{1, 3}] ])),
-  ?assertEqual(<<"test">>, get_series(Data)),
-  ?assertEqual([<<"1">>], get_columns(Data)),
-  ?assertEqual([[1], [3]], get_points(Data)).
+  ?assertEqual(
+    <<"test 1=1\ntest 1=3\n">>,
+    encode(test, [ [{1, 1}], [{1, 3}] ], [])
+  ).
 
 encode_2_test() ->
-  Data = decode_data(
-    encode("test", [ [{<<"1">>, 1}], [{<<"1">>, 3}] ])
-  ),
-  ?assertEqual(<<"test">>, get_series(Data)),
-  ?assertEqual([<<"1">>], get_columns(Data)),
-  ?assertEqual([[1], [3]], get_points(Data)).
+  ?assertEqual(
+    <<"test 1=1\ntest 1=3\n">>,
+    encode("test", [ [{<<"1">>, 1}], [{<<"1">>, 3}] ], [])
+  ).
 
 encode_3_test() ->
-  Data = decode_data(
-    encode(<<"test">>, [#{'1' => 1}, #{'1' => 3}])
-  ),
-  ?assertEqual(<<"test">>, get_series(Data)),
-  ?assertEqual([<<"1">>], get_columns(Data)),
-  ?assertEqual([[1], [3]], get_points(Data)).
+  ?assertEqual(
+    <<"test 1=1\ntest 1=3\n">>,
+    encode(<<"test">>, [#{'1' => 1}, #{'1' => 3}], [])
+  ).
 
 encode_4_test() ->
-  Data = decode_data(encode(<<"test">>, #{1 => 1})),
-  ?assertEqual(<<"test">>, get_series(Data)),
-  ?assertEqual([<<"1">>], get_columns(Data)),
-  ?assertEqual([[1]], get_points(Data)).
+  ?assertEqual(
+    <<"test 1=1\n">>,
+    encode(<<"test">>, #{1 => 1}, [])
+  ).
 
 encode_5_test() ->
-  ?assertNot(encode(<<"test">>, 'bla-bla')).
+  ?assertNot(encode(<<"test">>, 'bla-bla', [])).
+
+encode_with_tags_1_test() ->
+  ?assertEqual(
+    <<"test,host=eu-west,ip=1.1.1.1 1=1\n">>,
+    encode(<<"test">>, #{1 => 1}, [{host, "eu-west"}, {ip, '1.1.1.1'}])
+  ).
+
+encode_with_tags_2_test() ->
+  ?assertEqual(
+    <<"test,host=eu-west,ip=1.1.1.1 1=1\n">>,
+    encode(<<"test">>, #{1 => 1}, #{host => 'eu-west', ip => <<"1.1.1.1">>})
+  ).
+
+encode_with_tags_3_test() ->
+  ?assertEqual(
+    <<"test,host=eu-west,num=111 memory=\"high\",cpu=20\ntest,host=eu-west,num=111 memory=\"low\",cpu=30\n">>,
+    encode(
+      <<"test">>,
+      [
+        [{memory, "high"}, {"cpu", 20}],
+        [{<<"memory">>, "low"}, {cpu, 30}]
+      ], 
+      #{host => 'eu-west', num => 111}
+    )
+  ).
+
+encode_with_time_1_test() ->
+  ?assertEqual(
+    <<"test,host=eu-west,ip=1.1.1.1 1=1 123123\n">>,
+    encode(<<"test">>, #{1 => 1, time => 123123}, [{host, "eu-west"}, {ip, '1.1.1.1'}])
+  ).
+
+encode_with_time_2_test() ->
+  ?assertEqual(
+    <<"test,host=eu-west,ip=1.1.1.1 1=1,text=\"hello\" 123123\n">>,
+    encode(<<"test">>,
+      [[{"1", 1}, {text, <<"hello">>}, {time, 123123}]],
+      [{host, "eu-west"}, {ip, '1.1.1.1'}]
+    )
+  ).
+
+encode_escape_tags_test() ->
+  ?assertEqual(
+    <<"test,host=eu\\ we\\=st,ip=1\\,1\\,1\\,1 1=1 123123\n">>,
+    encode(<<"test">>,
+      [[{"1", 1}, {time, 123123}]],
+      [{host, "eu we=st"}, {ip, '1,1,1,1'}]
+    )
+  ).
+
+encode_escape_fields_test() ->
+  ?assertEqual(
+    <<"test,host=eu-west 1=1,text=\"hello\\ \\\"man\\=\\=human\\,\\ yo!\\\"\"\n">>,
```

Comment:
Line 396 is too long:     <<"test,host=eu-west 1=1,text=\"hello\\ \\\"man\\=\\=human\\,\\ yo!\\\"\"\n">>,.